### PR TITLE
feat: vagrant kong integration

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
## PR Details

PoC of Vagrant + Kong integration.

Frontend is now available through Kong at 192.168.33.60:8000

### CODE REVIEW

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -76,4 +77,18 @@ Vagrant.configure("2") do |config|
     node.vm.provision "register_dataplane", type: "shell", path: "common/register_dataplane.sh", env: {"KUMA_DATAPLANE_FILE" => "/vagrant/redis/kuma/dataplane.yaml", "KUMA_DATAPLANE_IP" => REDIS_IP}, privileged: false, run: "always"
   end
 
+  config.vm.define "kong" do |node|
+    node.vm.synced_folder "./kong/app/config", "/home/vagrant/kuma-demo/kong/"
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply vagrant/Vagrantfile]: removed

File: vagrant/Vagrantfile [hunk 2]
```diff
@@ -76,4 +77,18 @@ Vagrant.configure("2") do |config|
...
+    node.vm.box = "ubuntu/xenial64"
+    node.vm.hostname = "kong"
+    node.vm.network :private_network, ip: KONG_IP
+    node.vm.provision "update_etc_hosts", type: "shell", path: "common/update_etc_hosts.sh", env: { "KUMA_CONTROL_PLANE_IP" => KUMA_CONTROL_PLANE_IP }
+    node.vm.provision "install_kong", type: "shell", path: "kong/app/install.sh"
+    node.vm.provision "start_kong", type: "shell", path: "kong/app/start.sh", privileged: true, run: "always"
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply vagrant/Vagrantfile]: removed, no need to be explicit

File: vagrant/kong/app/start.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+cd /home/vagrant/kuma-demo/kong/
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply vagrant/kong/app/start.sh]: updated

File: vagrant/kong/app/config/config.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# it specifies the minimum version of Kong that supports the format
+
+_format_version: "1.1"
+
+# Each Kong entity (core entity or custom entity introduced by a plugin)
+# can be listed in the top-level as an array of objects:
+
+services:
+- name: frontend
+  url: http://localhost:3001
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply vagrant/kong/app/config/config.yaml]: updated

File: vagrant/kong/kuma/dataplane.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+type: Dataplane
+mesh: default
+name: kong
+networking:
+  inbound:
+  - interface: {{ DATAPLANE_IP }}:9876:9875 # doesn't matter since we will bypass it anyways
+    tags:
+      service: kong
+  outbound:
+  - interface: :3001
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply vagrant/kong/kuma/dataplane.yaml]: updated

File: vagrant/kong/kuma/dataplane.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+type: Dataplane
+mesh: default
+name: kong
+networking:
+  inbound:
```
Comment (yskopets) [inline vagrant/kong/kuma/dataplane.yaml]: Demo should be updated to use the new `gateway` field that has been introduced specifically for this scenario

**Pull request merged**